modules/elasticsearch/src/test/java/org/testcontainers/elasticsearch/ElasticsearchContainerTest.java

```diff
@@ -109,6 +109,16 @@ public void transportClientClusterHealth() {
         }
     }
 
+    @Test
+    public void waitingForShards() throws IOException{
+        try (ElasticsearchContainer container = new ElasticsearchContainer("jonvallet/elasticsearch").waitingForShards()) {
```

Comment:
I'm uneasy about adding a new personal image to our test suite like this - there are a few things:
* I can't find the source (Dockerfile/content) of this image
* As far as I can tell it doesn't share common layers with the official Elasticsearch images, meaning that this adds another ~480MB docker pull to dev/CI builds
* this image will use the latest tag, putting us at risk if you publish a new image with this tag in the future

Overall, I think it's probably better if we could replace this with a Dockerfile, based on the official ES image, that Testcontainers can [build on demand](https://www.testcontainers.org/usage/dockerfile.html). This way we don't create a dependency on your `jonvallet/elasticsearch` remaining identical for perpetuity.

Would you be able to share a Dockerfile? I'm happy to help if you need a hand setting this up with the [Dockerfile container](https://www.testcontainers.org/usage/dockerfile.html) model.

Reply:
I agree with you, I did not knew there was support for using a Dockerfile and do a build on demand, that is much safer that relying on my personal dockerhub account.

I have post an update that uses an ImageFromDockerfile instead. I had to expose the constructor from `GenericCointainer` to `ElasticsearchContainer` to be able to do it, let me know what you think about it.